pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -434,6 +434,10 @@ func (e *EndpointController) syncService(key string) error {
 		}
 	}
 
+	if currentEndpoints.Subsets == nil {
+		currentEndpoints.Subsets = []v1.EndpointSubset{}
```

Comment:
don't mutate this object... it is shared between controllers. prefer semantic.DeepEqual instead. you might also need to move up the check of whether this is a new object (if it is, we want to continue to persist it)

see a similar fix in https://github.com/kubernetes/kubernetes/pull/45294/commits/9f0b88a48d7224cbd074577592a788bade3f5133